src/main/java/edu/kit/provideq/toolbox/ProcessRunner.java

```diff
@@ -1,40 +1,220 @@
 package edu.kit.provideq.toolbox;
 
+import edu.kit.provideq.toolbox.meta.ProblemType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
 import java.io.BufferedReader;
+import java.io.File;
 import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
 
 /**
- * Runs a process.
+ * Runs a process to solve a problem.
+ * Accepts a ProblemBuilder
```

Comment:
What's a ProblemBuilder?

Reply:
The `ProblemBuilder` class from native Java which is used to build a process with commands or a working directory. 
We've used this already and I just included this in the documentation. Nothing has changed there. Or do you want to say that the documentation should be updated with a short explanation of `ProblemBuilder`?
Here's a link to the documentation: https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/lang/ProcessBuilder.html

Reply:
I think you're mixing up `ProblemBuilder` and `ProcessBuilder` here

Reply:
Ooh, yes I see. I'll change that.